Merge from Artem's fork

jbkempf opened the pull request:
This needs reviewing :)

Commit:
renamed md4/md5 context methods to prevent compiling issues with other libraries

Commit:
fixed rename request

Commit:
smb os and smblanman

Commit:
added SO_NOSIGPIPE

Commit:
xcode build project

Commit:
refresh project

Commit:
update project

Commit:
added missing files

Commit:
added block assertions

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	src/netbios_session.c

Commit:
deployment target

Commit:
Merge commit 'c3d186419708e52123137d23a3fe052ff78267ad'

Conflicts:
	.gitignore
	src/smb_session.h

Commit:
discard merge

Commit:
block assertions

Commit:
asn1_ calls synchronized

Commit:
clock ctime

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	Makefile.am
#	bin/dsm.c
#	include/bdsm.h
#	include/bdsm/smb_types.h
#	src/netbios_defs.h
#	src/netbios_ns.c
#	src/smb_file.c
#	src/smb_session.c
#	src/smb_types.h

rdmeleshko committed:
fixed compiling issues

rdmeleshko committed:
fixed asn lock

rdmeleshko committed:
merged 1ef10adc3b4f57d4fafc694227c6fcc49b88a0ca from libdsm

Commit:
merged last version 0.2.7 - removed local fixes

Commit:
md4 md5 rename

Commit:
CTX_ rename

Commit:
added build script

Commit:
Merge branch 'master' of https://github.com/leshkoapps/libdsm

Commit:
fixed clock gettime

Commit:
removed compiled lib

Commit:
update gitignore

Commit:
asn lock unlock

Commit:
fix rename

Commit:
minor

Commit:
smb_session_login_spnego: don't override logged status

It's already set by auth() in case of success.

Commit:
SMB SESSION LOGOFF

Commit:
asn synchronized fixes

Commit:
smb session assert - disabled for release

jbkempf commented:
Well, in fact, most of it is not good. But there are a couple of things to cherry-pick.

darling0825 committed:
fix bug "retutn" to "return"

darling0825 committed:
Update xcode setting

darling0825 committed:
fix a small bug

everappz committed:
Merge pull request #1 from darling0825/master

fix some small bug

Commit:
minor

Commit:
Merge branch 'master' of https://github.com/leshkoapps/libdsm

elthariel commented on .gitignore:
```diff
@@ -4,14 +4,12 @@ ref/*
 *.exe
 dsm
 *.so
-*.a
```
Duplicate, good catch

elthariel commented on .gitignore:
```diff
@@ -4,14 +4,12 @@ ref/*
 configure
 .*/
-build*/
-*.a
```
We still need to keep one of them

elthariel commented on .gitignore:
```diff
@@ -4,14 +4,12 @@ ref/*
 dsm_inverse
 configure
 .*/
-build*/
```
Hum, why ?

elthariel commented on .gitignore:
```diff
@@ -26,7 +24,12 @@ build*/
+config.s*
+config.log
+config.guess
+universal/*.a
```
Sure why not

elthariel reviewed (commented):
We definitely can't merge this as is :)

There might be interesting stuff in there but it's really hard to find out as there are a looot of indent changes.

I assume that what you wanted to get is the ios build support ?

jbkempf commented:
@elthariel no, most of the branch is very bad. There is an interesting patch about move(), but the rest is almost all useless...

elthariel commented:
There's also the asn1 thread safety part which is good

jbkempf commented:
@elthariel then we should cherry-pick those :)

elthariel commented:
Yeah. I started to have a look by but that won't be easy

On Mon, May 8, 2017, 17:41 Jean-Baptiste Kempf <notifications@github.com>
wrote:

> @elthariel <https://github.com/elthariel> then we should cherry-pick
> those :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/videolabs/libdsm/pull/105#issuecomment-299904782>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAJa866DdOKNBtDl4jlTmKawE8xbPmEfks5r3zfEgaJpZM4MVSR0>
> .
>

Commit:
fixed iconv crash and assert in release

Commit:
removed assert in release

Commit:
minor

Commit:
added ipv6 support

Commit:
ipv6 fixes

jbkempf commented:
@Kxuan do you now what we must cherry-pick?

Kxuan commented:
@jbkempf acturally, I dont known.. There are too many commits that I do not need. I just merge the 40c2404 to my fork...

jbkempf commented:
@Kxuan what is there in your fork? Just 40c2404 ?

Kxuan commented:
Yes. Only 4e226691e7359d798415dc09c7c901d9370ed666 is picked

chouquette commented:
`4e22669` has been merged

Commit:
added user port and connection timeout

Commit:
decreased connection timeout

Commit:
minor

Commit:
remove clock_gettime case of name conflict

Commit:
rename clock getttime to avoid name collision

Commit:
smb_ntlm: do not send lm & ntlm to server for guest login
smb_ntlm: Do not append empty buffers
smb_stat: add smb_stat_list_next()
Bump ABI revision

Commit:
smb_fstat: check malformed packet

Commit:
netbios: use time() instead of clock_gettime()

Commit:
Fix too small buffer for ASN1 response token.

* After having done a server upgrade to some newer version of samba,
smb_session_login_spnego() consistently failed. The reason was
that smb_spnego::challenge() invokes asn1_read_value() with a
buffer of size 256 bytes, but that function for some reason
requires a buffer of 258 bytes in size. Fix this the simple
way by increasing the on-stack buffer to 512 bytes.

Commit:
Fix setting of payload_size in smb_negotiate().

* On some platforms (arm5te apparently), writing to an unaligned
uint16_t pointer will actually write to the next lower word-aligned
address. Use memcpy() in order to fix this.

Commit:
Fixed segmentation fault

Commit:
minor

Commit:
asn and iconv lock refactoring

Commit:
minor

Commit:
added read/write timeout

Commit:
no message

Commit:
read timeout and write timeout refactoring

Commit:
bin/discover: fix netbios_ns leak

Commit:
minor

Commit:
fixed directory parsing for empty responses

Commit:
minor

Commit:
minor

Commit:
asn lock refactoring

Commit:
renamed bdsm headers

Commit:
Revert "renamed bdsm headers"

This reverts commit d55dfe95b7040314f98905d3055a7a18d015dec0.

Commit:
fixed headers

Commit:
debug header

Commit:
smbtypes

Commit:
headers

Commit:
headers

everappz committed:
Update smb_stat.h

Commit:
headers

Commit:
Merge branch 'master' of https://github.com/leshkoapps/libdsm

Commit:
headers

Commit:
minor

Commit:
minor

Commit:
headers

Commit:
headers

Commit:
headers

Commit:
fixed crash on TimeCapsule

Commit:
fixed assert in release

Commit:
minor

Commit:
minor

Commit:
added libtasn source code

Commit:
added tasn subproject

Commit:
minor

Commit:
minor

Commit:
minor

Commit:
minor

Commit:
fixed timeouts

Commit:
project file fixes

Commit:
minor

Commit:
fixed file info request

Commit:
added file info deprecated request/response

everappz committed:
fix deployment target

everappz committed:
project file fix

everappz committed:
project file fix

everappz committed:
added podspecs

everappz committed:
fixed podspec

everappz committed:
removed libtasn1 folder

everappz committed:
added libtasn1 as submodule

everappz committed:
podspec update

everappz committed:
podspec update

everappz committed:
podspec update

everappz committed:
fixed #include <libtasn1.h>
